### PR TITLE
Allow Matplotlib 1.5 usage (fix #978)

### LINKED ISSUE #978: requires Matplotlib 2?
7aca2b1584503c28dd9660d15f13630d9b690ef3 calls `set_facecolor`, which apparently was introduced in Matplotlib 2.0 (previously it was called [set_axis_bgcolor](https://matplotlib.org/api/_as_gen/matplotlib.axes.Axes.set_axis_bgcolor.html), I guess?)

Apparently this is causing problems [for users who still have Matplotlib 1.5](https://discourse.julialang.org/t/pyplot-keyerror-with-simple-plot-in-julia-0-6/4985?u=stevengj).

If you don't want to support Matplotlib 1.5, you should check `PyPlot.version` (this is the Matplotlib version) on startup and print an informative error message.

### PR DESCRIPTION
**Pull request by daschw**
I suppose we can allow the usage of Matplotlib v1.5 but give a hint that we no longer support it. I have not tested this with Matplotlib v1.5, but I think it should work.
still open to any suggestions ...

cc: @stevengj

### CODE REVIEW

File: src/backends/pyplot.jl [hunk 1]
```diff
@@ -83,6 +83,14 @@ function _initialize_backend(::PyPlotBackend)
         const pycollections = PyPlot.pywrap(PyPlot.pyimport("matplotlib.collections"))
         const pyart3d = PyPlot.pywrap(PyPlot.pyimport("mpl_toolkits.mplot3d.art3d"))
 
+        # "support" matplotlib v1.5
+        const set_facecolor_sym = if PyPlot.version < v"2"
+            warning("You are using Matplotlib $(PyPlot.version), which is no longer officialy supported by the Plots community. To ensure smooth Plots.jl integration update your Matplotlib library to a version >= 2.0.0")
```
Comment (stevengj) [inline src/backends/pyplot.jl]: Do you mean `warn("...")`?

Comment (daschw) [inline src/backends/pyplot.jl]: Yes, thank you!

**Comment by stevengj** [comment]
Has this been tested under Matplotlib 1.5?

**Comment by daschw** [comment]
No, is there a way to downgrade my Conda matplotlib for testing?

**Comment by stevengj** [comment]
Does something like `Conda.add("matplotlib=1.5.1")` work?

**Comment by daschw** [comment]
It works for downgrading, but when I do `using Plots; pyplot()` or `using PyPlot` afterwards, it's automatically updated again:
```julia
julia> using Conda

julia> Conda.add("matplotlib=1.5.1")
WARNING: Compat.UTF8String is deprecated, use String instead.
  likely near no file:0
Fetching package metadata .........
Solving package specifications: .

Package plan for installation in environment /home/dani/.julia/v0.6/Conda/deps/usr:

The following packages will be DOWNGRADED:

    matplotlib: 2.0.2-np111py27_0 --> 1.5.1-np111py27_0
    pyqt:       5.6.0-py27_2      --> 4.11.4-py27_4    
    qt:         5.6.2-4           --> 4.8.7-4          


julia> using PyPlot
INFO: Installing PyQt5 via the Conda pyqt package...
Fetching package metadata .........
Solving package specifications: .

Package plan for installation in environment /home/dani/.julia/v0.6/Conda/deps/usr:

The following packages will be UPDATED:

    matplotlib: 1.5.1-np111py27_0 --> 2.0.2-np111py27_0
    pyqt:       4.11.4-py27_4     --> 5.6.0-py27_2     
    qt:         4.8.7-4           --> 5.6.2-4          

/home/dani/.julia/v0.6/Conda/deps/usr/lib/python2.7/site-packages/matplotlib/__init__.py:1350: UserWarning:  This call to matplotlib.use() has no effect
because the backend has already been chosen;
matplotlib.use() must be called *before* pylab, matplotlib.pyplot,
or matplotlib.backends is imported for the first time.

  warnings.warn(_use_error_msg)
WARNING: No working GUI backend found for matplotlib
ERROR: InitError: PyError (ccall(@pysym(:PyImport_ImportModule), PyPtr, (Cstring,), name)) <type 'exceptions.ImportError'>
ImportError('cannot import name is_hashable',)
  File "/home/dani/.julia/v0.6/Conda/deps/usr/lib/python2.7/site-packages/matplotlib/pyplot.py", line 29, in <module>
    import matplotlib.colorbar
  File "/home/dani/.julia/v0.6/Conda/deps/usr/lib/python2.7/site-packages/matplotlib/colorbar.py", line 34, in <module>
    import matplotlib.collections as collections
  File "/home/dani/.julia/v0.6/Conda/deps/usr/lib/python2.7/site-packages/matplotlib/collections.py", line 37, in <module>
    import matplotlib.lines as mlines
  File "/home/dani/.julia/v0.6/Conda/deps/usr/lib/python2.7/site-packages/matplotlib/lines.py", line 18, in <module>
    from .cbook import (iterable, is_string_like, is_numlike, ls_mapper_r,

Stacktrace:
 [1] pyerr_check at /home/dani/.julia/v0.6/PyCall/src/exception.jl:56 [inlined]
 [2] pyerr_check at /home/dani/.julia/v0.6/PyCall/src/exception.jl:61 [inlined]
 [3] macro expansion at /home/dani/.julia/v0.6/PyCall/src/exception.jl:81 [inlined]
 [4] pyimport(::String) at /home/dani/.julia/v0.6/PyCall/src/PyCall.jl:370
 [5] __init__() at /home/dani/.julia/v0.6/PyPlot/src/init.jl:184
 [6] macro expansion at ./REPL.jl:97 [inlined]
 [7] (::Base.REPL.##1#2{Base.REPL.REPLBackend})() at ./event.jl:73
during initialization of module PyPlot
```

**Comment by mkborregaard** [comment]
I don't see anything wrong with this, but in principle I don't think we should suport back versions at this stage. The major difficulty with maintaining Plots is that it depends on other softwares that're developing fast themselves, and we've got no way of keeping track. Supporting back versions of those packages is likely to cause issues and divert from the more important task of developing Plots.

**Comment by daschw** [comment]
@mkborregaard Are you suggesting to return an informative error if matplotlib version < 2 and no longer allowing 1.5 at all when using Plots?

**Comment by stevengj** [comment]
@daschw, just prevent it from defaulting to the PyQt5 backend.  e.g. do `using PyCall; pygui(:tk)` before importing PyPlot.

**Comment by daschw** [comment]
still automatically updating:
```julia
julia> using Conda; Conda.add("matplotlib=1.5.1")
WARNING: Compat.UTF8String is deprecated, use String instead.
  likely near no file:0
Fetching package metadata .........
Solving package specifications: .

Package plan for installation in environment /home/dani/.julia/v0.6/Conda/deps/usr:

The following packages will be DOWNGRADED:

    matplotlib: 2.0.2-np111py27_0 --> 1.5.1-np111py27_0
    pyqt:       5.6.0-py27_2      --> 4.11.4-py27_4    
    qt:         5.6.2-4           --> 4.8.7-4          


julia> using PyCall; pygui(:tk)
:tk

julia> using PyPlot
INFO: Installing PyQt5 via the Conda pyqt package...
Fetching package metadata .........
Solving package specifications: .

Package plan for installation in environment /home/dani/.julia/v0.6/Conda/deps/usr:

The following packages will be UPDATED:

    matplotlib: 1.5.1-np111py27_0 --> 2.0.2-np111py27_0
    pyqt:       4.11.4-py27_4     --> 5.6.0-py27_2     
    qt:         4.8.7-4           --> 5.6.2-4          

ERROR: InitError: PyError (ccall(@pysym(:PyImport_ImportModule), PyPtr, (Cstring,), name)) <type 'exceptions.ImportError'>
ImportError('cannot import name is_hashable',)
  File "/home/dani/.julia/v0.6/Conda/deps/usr/lib/python2.7/site-packages/matplotlib/pyplot.py", line 29, in <module>
    import matplotlib.colorbar
  File "/home/dani/.julia/v0.6/Conda/deps/usr/lib/python2.7/site-packages/matplotlib/colorbar.py", line 34, in <module>
    import matplotlib.collections as collections
  File "/home/dani/.julia/v0.6/Conda/deps/usr/lib/python2.7/site-packages/matplotlib/collections.py", line 37, in <module>
    import matplotlib.lines as mlines
  File "/home/dani/.julia/v0.6/Conda/deps/usr/lib/python2.7/site-packages/matplotlib/lines.py", line 18, in <module>
    from .cbook import (iterable, is_string_like, is_numlike, ls_mapper_r,

Stacktrace:
 [1] pyerr_check at /home/dani/.julia/v0.6/PyCall/src/exception.jl:56 [inlined]
 [2] pyerr_check at /home/dani/.julia/v0.6/PyCall/src/exception.jl:61 [inlined]
 [3] macro expansion at /home/dani/.julia/v0.6/PyCall/src/exception.jl:81 [inlined]
 [4] pyimport(::String) at /home/dani/.julia/v0.6/PyCall/src/PyCall.jl:370
 [5] __init__() at /home/dani/.julia/v0.6/PyPlot/src/init.jl:184
 [6] _require_from_serialized(::Int64, ::Symbol, ::String, ::Bool) at ./loading.jl:200
 [7] _require_search_from_serialized(::Int64, ::Symbol, ::String, ::Bool) at ./loading.jl:236
 [8] _require(::Symbol) at ./loading.jl:434
 [9] require(::Symbol) at ./loading.jl:398
 [10] macro expansion at ./REPL.jl:97 [inlined]
 [11] (::Base.REPL.##1#2{Base.REPL.REPLBackend})() at ./event.jl:73
during initialization of module PyPlot
```

**Comment by stevengj** [comment]
Whoops, you have to do it via `ENV["MPLBACKEND"] = "tkagg"`

**Comment by daschw** [comment]
OK, thanks @stevengj, I managed to test it now with Matplotlib version 1.5.1.
```julia
using Conda; Conda.add("matplotlib=1.5.1")
ENV["MPLBACKEND"] = "tkagg"
Pkg.build("PyPlot")
using Plots; pyplot()

plot(rand(10), background_color_legend = :green, background_color_inside = :yellow, background_color_outside = :red)
```
works as expected.

Still, i think @mkborregaard made a valid point, so I'm not sure if this should be merged. Any thoughts?

**Comment by stevengj** [comment]
It seems reasonable to merge simple, small patches for 1.5 support, even if the main Plots developers don't want to worry about it themselves. 

If 1.5 support ever requires major restructuring, or if people who want it aren't willing to submit PRs, that will be the time to drop it

**Comment by mkborregaard** [comment]
I'm fine with that.

**Comment by daschw** [comment]
Great, I also agree.

**Pull request merged**